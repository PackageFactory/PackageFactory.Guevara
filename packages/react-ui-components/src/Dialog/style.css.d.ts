export interface IStyleCss {
  'dialog': string;
  'dialog__contentsPosition': string;
  'dialog__contents': string;
  'dialog--wide': string;
  'dialog--narrow': string;
  'dialog__title': string;
  'dialog__closeBtn': string;
  'dialog__body': string;
  'dialog__actions': string;
}

export const locals: IStyleCss;
