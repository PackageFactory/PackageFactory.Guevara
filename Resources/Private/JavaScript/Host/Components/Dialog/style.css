.dialog {
    position: fixed;
    top: 0;
    left: 0;
    z-index: 999;
    width: 100vw;
    height: 100vh;
    background: rgba(0,0,0, .35);
}
.dialog__contents {
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translateX(-50%) translateY(-50%);
    background: $darker
}
.dialog__contents__inner {
    position: relative;
    width: 80vw;
    min-height: 300px;
    padding: $spacing;
}
.dialog__contents__inner__closeBtn {
    position: absolute;
    top: 0;
    right: 0;
}
.dialog__contents__inner__actions {
    text-align: right;
}
