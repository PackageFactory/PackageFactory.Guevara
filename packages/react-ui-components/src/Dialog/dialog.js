import React from 'react';
import PropTypes from 'prop-types';
import mergeClassNames from 'classnames';
import omit from 'lodash.omit';
import Portal from 'react-portal';
import CloseOnEscape from 'react-close-on-escape';

const validStyleKeys = ['wide', 'narrow'];

export const DialogWithoutEscape = props => {
    const {
        className,
        title,
        style,
        children,
        isOpen,
        onRequestClose,
        actions,
        theme,
        IconButtonComponent,
        ...restProps
    } = props;
    const rest = omit(restProps, ['isOpen']);
    const finalClassName = mergeClassNames({
        [theme.dialog]: true,
        [theme['dialog--wide']]: style === 'wide',
        [theme['dialog--narrow']]: style === 'narrow',
        [className]: className && className.length
    });

    return (
        <Portal isOpened={isOpen}>
            <section {...rest} className={finalClassName} role="dialog" tabIndex="0">
                <div className={theme.dialog__contentsPosition}>
                    <div className={theme.dialog__contents}>
                        {onRequestClose &&
                        <IconButtonComponent
                            icon="close"
                            className={theme.dialog__closeBtn}
                            onClick={onRequestClose}
                            />
                          }
                        <div className={theme.dialog__title}>
                            {title}
                        </div>

                        <div className={theme.dialog__body}>
                            {children}
                        </div>

                        {actions && actions.length ?
                            <div className={theme.dialog__actions}>
                                {React.Children.map(actions, (action, index) => <span key={index}>{action}</span>)}
                            </div> : null
                        }
                    </div>
                </div>
            </section>
        </Portal>
    );
};
DialogWithoutEscape.propTypes = {
    /**
     * This prop controls the rendered state of the Dialog, when falsy, nothing gets rendered into the DOM.
     */
    isOpen: PropTypes.bool.isRequired,

    /**
     * The handler which gets called once the user clicks on the close symbol in the top right corner of the Dialog.
     */
    onRequestClose: PropTypes.func.isRequired,

    /**
     * The title to be rendered on top of the Dialogs contents.
     */
    title: PropTypes.any,

    /**
     * The `style` prop defines the visual style of the `Dialog`.
     */
    style: PropTypes.oneOf(validStyleKeys),

    /**
     * The contents to be rendered within the Dialog.
     */
    children: PropTypes.any.isRequired,

    /**
     * An Array of nodes(e.g. Action Buttons) which are placed at the bottom of the Dialog.
     */
    actions: PropTypes.any,

    /**
     * An optional `className` to attach to the wrapper.
     */
    className: PropTypes.string,

    /**
     * An optional `contentsClassName` to attach to the content area of the dialog.
     */
    contentsClassName: PropTypes.string,

    /**
     * An optional css theme to be injected.
     */
    theme: PropTypes.shape({
        'dialog': PropTypes.string,
        'dialog__contentsPosition': PropTypes.string,
        'dialog__contents': PropTypes.string,
        'dialog__title': PropTypes.string,
        'dialog__closeBtn': PropTypes.string,
        'dialog__actions': PropTypes.string,
        'dialog--wide': PropTypes.string,
        'dialog--narrow': PropTypes.string
    }).isRequired,

    /**
     * Static component dependencies which are injected from the outside (index.js)
     */
    IconButtonComponent: PropTypes.any.isRequired
};

const DialogWithEscape = props => {
    const onEscape = () => props.onRequestClose();
    return (
        <CloseOnEscape onEscape={onEscape}><DialogWithoutEscape {...props}/></CloseOnEscape>
    );
};
DialogWithEscape.propTypes = {
    onRequestClose: PropTypes.func.isRequired
};

export default DialogWithEscape;
