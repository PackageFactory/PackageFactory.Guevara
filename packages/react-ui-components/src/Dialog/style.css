.dialog {
    composes: reset from './../reset.css';
    position: fixed;
    top: 0;
    left: 0;
    z-index: var(--zIndex-Dialog-Context);
    width: 100vw;
    height: 100vh;
    background: rgba(#000, .35);
}
.dialog__contentsPosition {
    composes: reset from './../reset.css';
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translateX(-50%) translateY(-50%);
    background: var(--brandColorsContrastDarker);
}
.dialog__contents {
    composes: reset from './../reset.css';
    position: relative;
    width: calc(100vw - var(--goldenUnit) * 2);
    max-width: calc(var(--goldenUnit) * 16);
    border: 1px solid var(--brandColorsContrastDark);

    .dialog--wide & {
        max-width: calc(var(--goldenUnit) * 24);
    }

    .dialog--narrow & {
        max-width: 400px;
    }
}
.dialog__title {
    composes: reset from './../reset.css';
    font-size: 16px;
    line-height: 1.2;
    padding: var(--spacing);
}
.dialog__closeBtn {
    composes: reset from './../reset.css';
    position: absolute;
    top: 0;
    right: 0;
}
.dialog__body {
    max-height: calc(65vh);
    overflow-y: auto;
}
.dialog__actions {
    composes: reset from './../reset.css';
    text-align: right;
    border-top: 1px solid var(--brandColorsContrastDark);
    padding: var(--spacing);
}
.dialog__actions > * {
    margin-left: calc(var(--spacing) / 2);
}
