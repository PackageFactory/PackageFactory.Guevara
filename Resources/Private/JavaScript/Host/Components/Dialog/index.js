import React, {PropTypes} from 'react';
import mergeClassNames from 'classnames';
import {executeCallback} from 'Host/Abstracts/';
import IconButton from 'Host/Components/IconButton/';
import Portal from 'Host/Components/Portal/';
import style from './style.css';

const Dialog = props => {
    const {
        className,
        children,
        isOpen
    } = props;
    const classNames = mergeClassNames({
        [style.dialog]: true,
        [className]: className && className.length
    });

    return (
        <Portal targetId="dialog" isOpened={isOpen}>
            <section className={classNames}>
                <div className={style.dialog__contents}>
                    <div className={style.dialog__contents__inner}>
                        <IconButton
                            icon="close"
                            className={style.dialog__contents__inner__closeBtn}
                            onClick={e => executeCallback(e, props.onRequestClose)}
                            />

                        {children}

                        <div className={style.dialog__contents__inner__actions}>
                            {props.actions.map((action, index) => <span key={index}>{action}</span>)}
                        </div>
                    </div>
                </div>
            </section>
        </Portal>
    );
};
Dialog.propTypes = {
    // State propTypes.
    isOpen: PropTypes.bool.isRequired,

    // Will be called once the close icon in the top right corner gets clicked.
    onRequestClose: PropTypes.func.isRequired,

    // Contents of the Dialog.
    children: PropTypes.node.isRequired,

    // Optional Array of nodes(Action buttons f.e.) which are placed at the bottom of the Dialog.
    actions: PropTypes.node,

    // Style related propTypes.
    className: PropTypes.string
};

export default Dialog;
