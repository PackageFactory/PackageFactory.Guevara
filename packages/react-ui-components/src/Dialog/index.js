import {themr} from '@friendsofreactjs/react-css-themr';
import identifiers from './../identifiers';
import style from './style.css';
import Dialog from './dialog.js';

const ThemedDialog = themr(identifiers.dialog, style)(Dialog);

//
// Dependency injection
//
import injectProps from './../_lib/injectProps.js';
import IconButton from './../IconButton/index';

export default injectProps({
    IconButtonComponent: IconButton
})(ThemedDialog);
