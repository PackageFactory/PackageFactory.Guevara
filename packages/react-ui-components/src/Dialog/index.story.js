import React from 'react';
import {storiesOf, action} from '@storybook/react';
import {withKnobs, text, boolean} from '@storybook/addon-knobs';
import {StoryWrapper} from './../_lib/storyUtils.js';
import Dialog from './index.js';
import Button from './../Button/index.js';

storiesOf('Dialog', module)
    .addDecorator(withKnobs)
    .addWithInfo(
        'default',
        'Dialog',
        () => (
            <StoryWrapper>
                <Dialog
                    isOpen={boolean('Is opened?', true)}
                    title={text('Title', 'Hello title!')}
                    onRequestClose={action('onRequestClose')}
                    actions={[
                        <Button key="foo">An action button</Button>
                    ]}
                    style="wide"
                    >
                    {text('Inner content', 'Hello world!')}
                </Dialog>
            </StoryWrapper>
        ),
        {inline: true, source: false}
    )
    .addWithInfo(
        'narrow',
        'Dialog',
        () => (
            <StoryWrapper>
                <Dialog
                    isOpen={boolean('Is opened?', true)}
                    title={text('Title', 'Hello title!')}
                    onRequestClose={action('onRequestClose')}
                    actions={[
                        <Button key="foo">An action button</Button>
                    ]}
                    style="narrow"
                    >
                    {text('Inner content', 'Hello world!')}
                </Dialog>
            </StoryWrapper>
        ),
        {inline: true, source: false}
    );
