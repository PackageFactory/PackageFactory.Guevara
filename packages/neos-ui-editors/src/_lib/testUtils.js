import {PureComponent, Children} from 'react';
import PropTypes from 'prop-types';
import {SynchronousRegistry, SynchronousMetaRegistry} from '@neos-project/neos-ui-extensibility/src/registry';

const globalRegistry = new SynchronousMetaRegistry();

// i18n Registry
class FakeI18NRegistry extends SynchronousRegistry {
    translate(key) {
        return key;
    }
}
globalRegistry.set('i18n', new FakeI18NRegistry());

// Data Source DataLoader
export const MockDataSourceDataLoader = {
    _currentPromise: null,
    _currentPromiseResolveFn: null,

    // API in tests
    reset() {
        this._currentPromise = null;
    },

    resolveCurrentPromise(result) {
        this._currentPromiseResolveFn(result);
        return this._currentPromise;
    },

    // API of DataLoader
    resolveValue() {
        if (!this._currentPromise) {
            this._currentPromise = new Promise(resolve => {
                this._currentPromiseResolveFn = resolve;
            });
        }
        return this._currentPromise;
    }
};
globalRegistry.set('dataLoaders', new SynchronousRegistry());
globalRegistry.get('dataLoaders').set('DataSources', MockDataSourceDataLoader);

// wrapper
export class WrapWithMockGlobalRegistry extends PureComponent {
    static propTypes = {
        children: PropTypes.element.isRequired
    };

    static childContextTypes = {
        globalRegistry: PropTypes.object.isRequired,
        configuration: PropTypes.object.isRequired
    };

    getChildContext() {
        const configuration = {};
        return {globalRegistry, configuration};
    }

    render() {
        return Children.only(this.props.children);
    }
}
