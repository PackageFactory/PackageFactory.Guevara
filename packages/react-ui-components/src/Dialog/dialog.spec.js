import sinon from 'sinon';
import {createShallowRenderer, createStubComponent} from './../_lib/testUtils.js';
import Dialog from './dialog.js';
import Portal from 'react-portal';

const IconButtonComponent = createStubComponent();
const defaultProps = {
    isOpen: false,
    theme: {
        'dialog--wide': 'wideClassName'
    },
    children: 'Foo children',
    actions: ['Foo 1', 'Foo 2'],
    onRequestClose: () => null,
    IconButtonComponent
};
const shallow = createShallowRenderer(Dialog, defaultProps);

test('should render a "Portal" as the wrapping Component.', () => {
    const portal = shallow().find(Portal);

    expect(portal.length).toBe(1);
});
test('should pas a falsy "isOpened" tag to the "Portal" component if the "isOpen" prop is falsy.', () => {
    const portal = shallow().find(Portal);

    expect(portal.prop('isOpened')).toBe(false);
});
test('should pas a truthy "isOpened" tag to the "Portal" component if the "isOpen" prop is truthy.', () => {
    const portal = shallow({isOpen: true}).find(Portal);

    expect(portal.prop('isOpened')).toBe(true);
});
test('should render a "section" inside the portal with the attribute role="dialog".', () => {
    const portal = shallow().find(Portal);
    const section = portal.find('section');

    expect(section.length).toBe(1);
    expect(section.html().includes('role="dialog"')).toBeTruthy();
});
test('should render the "className" prop if passed.', () => {
    const portal = shallow({
        className: 'barClassName'
    }).find(Portal);
    const section = portal.find('section');

    expect(section.hasClass('barClassName')).toBeTruthy();
});
test('should render the "dialog--wide" className from the "theme" prop if the "isWide" prop is truthy.', () => {
    const portal = shallow({
        isWide: true
    }).find(Portal);
    const section = portal.find('section');

    expect(section.hasClass('wideClassName')).toBeTruthy();
});
test('should render the actions if passed.', () => {
    const portal = shallow().find(Portal);
    const section = portal.find('section');

    expect(section.html().includes('Foo 1')).toBeTruthy();
    expect(section.html().includes('Foo 2')).toBeTruthy();
});
test('should call the "onRequestClose" prop when clicking on the "IconButtonComponent" component.', () => {
    const onRequestClose = sinon.spy();
    const portal = shallow({onRequestClose}).find(Portal);
    const btn = portal.find(IconButtonComponent);

    btn.simulate('click');

    expect(onRequestClose.calledOnce).toBeTruthy();
});
