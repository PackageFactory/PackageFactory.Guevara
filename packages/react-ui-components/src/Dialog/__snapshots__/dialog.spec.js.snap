// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dialog/> should render correctly. 1`] = `
<Portal
  isOpened={false}
  onClose={[Function]}
  onOpen={[Function]}
  onUpdate={[Function]}
>
  <section
    className=""
    role="dialog"
    tabIndex="0"
  >
    <div>
      <div>
        <Component
          icon="times"
          onClick={[Function]}
        />
        <div />
        <div>
          Foo children
        </div>
        <div>
          <span
            key="0/.0"
          >
            Foo 1
          </span>
          <span
            key="1/.1"
          >
            Foo 2
          </span>
        </div>
      </div>
    </div>
  </section>
</Portal>
`;
